### PR TITLE
Docs: Add health-check command

### LINKED ISSUE #1542: Expand beamer command documentation
With the recent addition of the toplevel `beamer` command there is a need to reorganize our command docs, as well as add new docs for the `beamer health-check` command.

Ideally, we would have a "command reference", starting with the toplevel `beamer` command and then sections for individual commands (currently just `agent` and `health-check`). Agent configuration file docs are currently combined with command-line options docs so we would need to split them and move the config file docs into a separate sections, with links between the two.

Tasks:
- [x] introduce a command reference chapter
- [x] introduce a configuration files chapter
- [x] move existing `beamer-agent` docs into a standalone section within the command reference chapter
- [x] move and rework agent config file part into its own section within the config files chapter
- [x] document the `beamer health-check` command
- [x] document the `beamer health-check` config file within the config files chapter

### PR DESCRIPTION
**Pull request by manuelwedler**
Closes #1542

### CODE REVIEW

File: docs/source/commands.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+.. _command-health-check:
+
+``beamer health-check``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``health-check`` command scans the contracts for the emitted events and 
+analyzes whether there is a missed fill, unclaimed transaction or a challenge 
+game going on. In addition to this if an agent address is provided in the config 
+file it will fetch that agent's balance from the chain. The command will notify
+the user by sending everything either to Telegram or RocketChat.
```
Comment (GabrielBuragev) [inline docs/source/commands.rst:72]: >In addition to this if an agent address is provided in the config 
file it will fetch that agent's balance from the chain.

Can we rephrase this to:
"In addition to this, if an `agent-address` is provided in the config file, the final notification will also include the liquidity on all chains for all the tokens specified inside the configuration file for the provided agent address."

With this, I would like to point out that the monitoring script is useful for tracking agents liquidity on all chains for all tokens they support.

**Review by GabrielBuragev — APPROVED** [review]
Two minor suggestions on improvement. 
Other then that looks good, thanks!

File: docs/source/commands.rst [hunk 2]
```diff
@@ -0,0 +1,58 @@
+Command reference
+-----------------
+
+
+
+
```
Comment (istankovic) [inline docs/source/commands.rst:9]: Nitpick: seems like 2 extra empty lines are here, compared to what we have elsewhere.

File: docs/source/commands.rst [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+       Default: ``info``.
+
+   * - ``--metrics-prometheus-port PORT``
+     - Provide Prometheus metrics on the specified port.
+
+   * - ``--source-chain NAME``
+     - Name of the source chain. Deprecated and will be removed.
+       No longer needed because the agent supports multiple chain pairs.
+
+
```
Comment (istankovic) [inline docs/source/commands.rst:51]: Nitpick: extra empty line.

File: docs/source/configuration.rst [hunk 4]
```diff
@@ -117,103 +114,233 @@ address ``0x2644292EE5aed5c17BDcc6EDF1696ba802351cf6``, while the TST token cont
...
+
+        [chains.foo]
+        rpc-url = "http://foo.bar:8545"
+
+
+.. _config-health-check:
+
+Health Check configuration
+--------------------------
+
```
[the remark this answers is not in the document]
Comment (manuelwedler) [reply docs/source/configuration.rst:205]: Sounds like a great idea! By migrate you mean to also remove it from the run-your-own-agent README?

[the remark this answers is not in the document]
Comment (GabrielBuragev) [reply docs/source/configuration.rst:205]: > By migrate you mean to also remove it from the run-your-own-agent README?

In my opinion that repository, and its documentation respectively, is supposed to help users spin up agents in a quick manner and briefly explain the command possibilities / options that will get them up and running. The detailed documentation for those commands should be stored in our main docs (here) and use link references in the run-your-own-agent repo. Therefore, I think the telegram notification system docs should not reside in the run-your-own-agents README file.

This way we will also avoid spreading duplicate content around.

@istankovic What are your thoughts?

[the remark this answers is not in the document]
Comment (istankovic) [reply docs/source/configuration.rst:205]: > In my opinion that repository, and its documentation respectively, is supposed to help users spin up agents in a quick manner and briefly explain the command possibilities / options that will get them up and running. The detailed documentation for those commands should be stored in our main docs (here) and use link references in the run-your-own-agent repo. Therefore, I think the telegram notification system docs should not reside in the run-your-own-agents README file.
> 
> This way we will also avoid spreading duplicate content around.
> 
> @istankovic What are your thoughts?

Exactly what you said. It was never the intention of `run-your-own-agent` repo to provide docs, it was rather intended as "here are a few quick steps you can follow to get up and running; for anything more than that, please see our docs".

[the remark this answers is not in the document]
Comment (GabrielBuragev) [reply docs/source/configuration.rst:205]: Alright, glad we are on the same page.
So, @manuelwedler feel free to migrate that part of the docs or at least leave a comment under that PR so whoever is trying to merge it adjusts it before doing so.

File: docs/source/commands.rst (hunk 2 above)
Comment (manuelwedler) [reply docs/source/commands.rst:8]: :heavy_check_mark:

**Review by GabrielBuragev — APPROVED** [review]
Pefect, thanks for addressing the comments!
